drivers/spi/spi_sam0.c

```diff
@@ -411,14 +422,225 @@ static int spi_sam0_transceive_sync(struct device *dev,
 }
 
 #ifdef CONFIG_SPI_ASYNC
+
+static void spi_sam0_dma_rx_done(void *arg, u32_t id, int error_code);
+
+static int spi_sam0_dma_rx_load(struct device *dev, u8_t *buf,
+				size_t len)
+{
+	const struct spi_sam0_config *cfg = dev->config->config_info;
+	struct spi_sam0_data *data = dev->driver_data;
+	SercomSpi *regs = cfg->regs;
+	struct dma_config dma_cfg = {0};
+	struct dma_block_config dma_blk = {0};
+	int retval;
+
+	dma_cfg.channel_direction = PERIPHERAL_TO_MEMORY;
+	dma_cfg.source_data_size = 1;
+	dma_cfg.dest_data_size = 1;
+	dma_cfg.callback_arg = dev;
+	dma_cfg.dma_callback = spi_sam0_dma_rx_done;
+	dma_cfg.block_count = 1;
+	dma_cfg.head_block = &dma_blk;
+	dma_cfg.dma_slot = cfg->rx_dma_request;
+
+	dma_blk.block_size = len;
+	if (buf != NULL) {
+		dma_blk.dest_address = (u32_t)buf;
+	} else {
+		static u8_t dummy;
+
+		dma_blk.dest_address = (u32_t)&dummy;
+		dma_blk.dest_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+	}
+	dma_blk.source_address = (u32_t)(&(regs->DATA.reg));
```

Comment:
Add an empty line after a '}' (unless it's another '}').

Apply that everywhere.

Reply:
Changed and rebased for SAMR21 support.